### DISCUSSION TITLE
Did I do the legal stuff right?

### DISCUSSION THREAD

**Discussion by JC3** [assimp/assimp#3769] [Q&A] [answered]
I have an application; open source, free, non-commercial, released under MIT license. It uses libassimp and the assimp DLLs are included in the distribution. It has an installer on Windows. I've also got assimp binaries -- not source -- that I've built sitting in the source repo.

I have:
- Included assimp's LICENSE file alongside mine in the source repo, both are installed by the binary installer.
- Included assimp's CREDITS file in the source repo, it is also installed by the binary installer (I'm pretty sure this isn't required, but it seems polite).
- Required the user to accept the terms of assimp's LICENSE file, as well as mine, in the installer. They are two separate pages in the installer.
- Added assimp (and poly2tri) copyright info to general documentation in the repo, which is also installed.

Have I covered all the bases? In particular, do I need to display assimp version and copyright info at runtime wherever I display my app's version? Or is it enough to just include everything in the docs and have the user accept the license during install?

Thanks!

**Comment by krishty** [comment]
Here’s what I worked out:
```
Libary          License	       include in bina-   include in sour-
                               -ry distribution?  ce distribution?

Assimp          MIT-like        yes                yes
Cineware SDK    ?               ?                  ?
Clipper         Boost           no                 yes
Open3DGC        MIT             yes                yes
OpenDDL-Parser  MIT             yes                yes
Poly2Tri        MIT-like        yes                yes
pugixml         MIT             yes                yes
RapidJSON       MIT             yes                yes
stb_image.h     Public Domain   no                 no
Unzip           zlib            appreciated        appreciated
utf8cpp         Boost           no                 yes
zlib            zlib            appreciated        appreciated
```
As you can see I don’t use all of the libraries, didn’t bother checking the other ones.

**Boost**
License needs only be included in source code, because [the 2nd paragraph](https://www.boost.org/LICENSE_1_0.txt) says _“… must be included in all copies of the Software … unless such copies or derivative works are solely in the form of machine-executable object code generated by a source language processor.”_

**zlib**
Needs oonly be included in the source code if the authors cannot be misinterpreted:
_“If you use this software in a product, an acknowledgment in the product documentation would be appreciated but is not required.”_

**MIT**
License must be included in both binary and source: _“The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.”_
(This is where Boost and MIT differ.)

**MIT-like**
Assimp and Poly2Tri are like MIT but with some extra bullet points regarding other stuff.

**Cineware**
(For .C4D) has its own license, and I couldn’t spot a word on the issue.

Sorry for not answering your specific question, but I hope it helps anyway!

**P.S.:** I’m not a lawyer, just someone on the internet etc.

**Comment by kimkulling** [comment, marked as answer]
From my point of view you are on a good track. Most of the libs we are using were already checked by me,
